Dockerfile

```diff
@@ -19,46 +46,41 @@ ENV PYTHONUNBUFFERED=1\
     PIP_NO_CACHE_DIR=off \
     # Save runtime because it is not look for updating pip version
     PIP_DISABLE_PIP_VERSION_CHECK=on \
-    PIP_DEFAULT_TIMEOUT=100
-
-USER root
+    PIP_DEFAULT_TIMEOUT=100 \
+    # Disable poetry interaction
+    POETRY_NO_INTERACTION=1 \
+    POETRY_VIRTUALENVS_IN_PROJECT=1 \
+    POETRY_VIRTUALENVS_CREATE=1 \
+    POETRY_CACHE_DIR=/tmp/poetry_cache
 
 WORKDIR /app
 
-ENV PYTHONPATH "${PYTHONPATH}:/app"
-
-RUN set -ex \
-    # Create a non-root user
-    && addgroup --system --gid 1001 appgroup \
-    && adduser --system --uid 1001 --gid 1001 --no-create-home appuser
+# Install Poetry. Fix poetry version to avoid breaking changes
+RUN pip install poetry==1.5.1
 
-    # Upgrade the package index and install security upgrades
-RUN --mount=type=cache,target=/var/lib/apt/lists \
-    --mount=type=cache,target=/var/cache,sharing=locked \
-    apt-get update \
-    && apt-get upgrade --assume-yes \
-    && apt-get install --assume-yes --no-install-recommends python3-pip
+# Install the app. Just copy the files needed to install the dependencies
+COPY pyproject.toml poetry.lock ./
 
-# Install dependencies
-FROM python-base as poetry
-
-WORKDIR /app
-# that contains poetry=<version> to install a specific version of Poetry.
-COPY requirements.txt ./
-RUN --mount=type=cache,target=/root/.cache python3.11 -m pip install --disable-pip-version-check --requirement=requirements.txt
+#Poetry cache is used to avoid installing the dependencies every time the code changes, we will keep this folder in development environment and remove it in production
+# --no-root, poetry will install only the dependencies avoiding to install the project itself, we will install the project in the final layer
+# --without dev to avoid installing dev dependencies, we do not need test and linters in production environment
+# --with dev to install dev dependencies, we need test and linters in development environment
+# --mount, mount a folder for plugins with poetry cache, this will speed up the process of building the image
+RUN  if [ $DEV ]; then \
+    echo "Installing dev dependencies"; \
+      --mount=type=cache,target=$POETRY_CACHE_DIR  poetry install --with dev --no-root; \
+    else \
+    echo "Skipping dev dependencies"; \
+      --mount=type=cache,target=$POETRY_CACHE_DIR  poetry install --without dev --no-root && rm -rf $POETRY_CACHE_DIR; \
+   fi
 
-# Do the conversion
-COPY poetry.lock pyproject.toml ./
-RUN poetry export --output=requirements.txt
+#setup our final runtime layer
+FROM base as runtime
 
-FROM python-base as development
-
-RUN --mount=type=cache,target=/root/.cache --mount=type=bind,from=poetry,source=/app,target=/poetry python3.11 -m pip install --disable-pip-version-check --no-deps --requirement=/poetry/requirements.txt
-
-WORKDIR /app
+#COPY --from=builder ${VIRTUAL_ENV} ${VIRTUAL_ENV}
 
 COPY . .
 
-CMD ["python3"]
+WORKDIR /app
 
-USER appuser
+CMD ["python3"]
```

Comment:
Maybe we change this here so it uses the `CMD ["tail", "-f", "/dev/null"]` like the Ruby boilerplate? Right now the container starts up and enters an interactive Python session, but you can't really use it (though you can `exec` into the container and start your own).

https://github.com/mlibrary/ruby-starter/blob/a2e23c38faa1ec9e498ea434d7038da2646bf94b/Dockerfile#L23

Reply:
I agree